src/errors.ts

```diff
@@ -0,0 +1,59 @@
+import { GraaspErrorDetails, GraaspError } from 'graasp';
+import { StatusCodes } from 'http-status-codes';
+
+export class GraaspValidationError implements GraaspError {
+  name: string;
+  code: string;
+  message: string;
+  statusCode?: number;
+  data?: unknown;
+  origin: 'plugin' | string;
+
+  constructor({ code, statusCode, message }: GraaspErrorDetails, data?: unknown) {
+    this.name = code;
+    this.code = code;
+    this.message = message;
+    this.statusCode = statusCode;
+    this.data = data;
+    this.origin = 'plugin';
+  }
+}
+
+export class InvalidFileItemError extends GraaspValidationError {
+  constructor(data?: unknown) {
+    super(
+      {
+        code: 'GPIZERR003',
```

Comment:
The code actually follow the plugin name, here you pasted `Graasp Plugin Import Zip ERRor` so I think you should go with `GPVERR`